# IMPORTAÇÃO DE BIBLIOTECAS E DRIVE

As 4 primeiras linhas foram adicionadas antes de saber o que seria necessário durante a execução do TCC. As seguintes foram acrescentadas assim que necessário.

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import tensorflow as tf
from keras.models import Sequential
from keras.layers import Dense, Activation
from sklearn.preprocessing import StandardScaler
from sklearn.svm import SVR
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_absolute_error,mean_squared_error
from sklearn.linear_model import ElasticNetCV

Montagem do Google Drive

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


# IMPORTAÇÃO DE DADOS

In [3]:
df_pre = pd.read_csv("/content/drive/MyDrive/tcc/dataframes/df_pre.csv")

# EXPLORAÇÃO DE DADOS

In [4]:
df_pre.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5570 entries, 0 to 5569
Data columns (total 34 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   codigo                5570 non-null   int64  
 1   aids                  5570 non-null   int64  
 2   animais               5570 non-null   int64  
 3   dengue                5570 non-null   int64  
 4   hanseniase            5570 non-null   int64  
 5   hepatites             5570 non-null   int64  
 6   sadquirida            5570 non-null   int64  
 7   scongenita            5570 non-null   int64  
 8   sgestante             5570 non-null   int64  
 9   toxico                5570 non-null   int64  
 10  tuberculose           5570 non-null   int64  
 11  autoprovocada_sim     5570 non-null   int64  
 12  autoprovocada_não     5570 non-null   int64  
 13  violencia             5570 non-null   int64  
 14  estabelecimentos      5570 non-null   int64  
 15  profissionais        

In [5]:
df_pre["estado"] = df_pre["estado"].astype("category")
df_pre["metropolitana"] = df_pre["metropolitana"].astype("category")
df_pre["amazonia"] = df_pre["amazonia"].astype("category")
df_pre["semiarido"] = df_pre["semiarido"].astype("category")
df_pre["regiao"] = df_pre["regiao"].astype("category")

In [6]:
df_pre.head(100)

codigo  aids  animais  dengue  hanseniase  hepatites  sadquirida  \
0   110001     1      192     833          51         41          28   
1   110037     1      199      90          86         12          13   
2   110040     5       62      65          43         66          18   
3   110034     5       78     126          24          8          16   
4   110002   121      252     198         224        240         296   
..     ...   ...      ...     ...         ...        ...         ...   
95  130200     5       57       2          12          1           5   
96  130210     3       39      17           4          4           5   
97  130230     6      141       3          33         75          55   
98  130240    10      312      60          71         79          78   
99  130250    63      412      94          58        101         166   

    scongenita  sgestante  toxico  ...  impostos        pib  per_capita  \
0            2         12     186  ...   29351.0   495629.0    21600.73   
1            1          2      31  ...   11248.0   288681.0    21802.07   
2            2         11      18  ...   18150.0   358464.0    16728.75   
3            1          7      35  ...   10423.0   260348.0    18065.91   
4            8         89     321  ...  284398.0  2578830.0    23908.38   
..         ...        ...     ...  ...       ...        ...         ...   
95           0          6       1  ...    4646.0   129557.0    14162.35   
96           2          5       1  ...    1378.0    64130.0    23277.54   
97           0         16       0  ...    6188.0   227757.0    15908.16   
98           9         41       5  ...   18852.0   506494.0    10994.24   
99          34         78      96  ...   51036.0  1451150.0    14902.39   

    populacao  latitude  longitude regiao nascimentos mortalidade_geral  \
0       22945 -11.92870   -61.9958      N        1828               659   
1       13241 -12.14050   -61.8379      N         803               252   
2       21428  -9.71470   -63.3193      N        1154               369   
3       14411 -11.34670   -62.2852      N         911               414   
4      107863  -9.90612   -63.0330      N        8580              2557   
..        ...       ...        ...    ...         ...               ...   
95       9148  -2.74119   -58.0297      N         844               185   
96       2755  -1.88274   -66.9295      N         752                77   
97      14317  -2.74904   -66.7696      N        3190               336   
98      46069  -7.26453   -64.7952      N        3723               776   
99      97377  -3.29104   -60.6220      N        9974              2061   

   mortalidade_infantil  
0                    23  
1                     4  
2                    24  
3                     8  
4                   111  
..                  ...  
95                    9  
96                   15  
97                   71  
98                   75  
99                  132  

[100 rows x 34 columns]

Sífilis adquirida não inclui sífilis em gestante porque alguns valores em gestante é maior do que adquirida. Explorei até achar no index 6 (codigo 110003) e para confirmar index 16 (codigo 110008).

In [7]:
df_sif = df_pre

Criação de um indicador chamado "squalidade", que representa a razão entre o total de casos de sífilis adquirida e o número de casos de sífilis em gestante. A ideia é que quanto maior for o valor da razão melhor é a a notificação naquela cidade, considerando que o número de casos totais em homens e mulheres não grávidas deveria ser muito maior do que o número de casos confirmados em mulheres grávidas por simples exercício de lógica.

In [8]:
df_sif["squalidade"] = df_sif["sadquirida"]/df_sif["sgestante"]

In [9]:
df_sif["squalidade"]

0       2.333333
1       6.500000
2       1.636364
3       2.285714
4       3.325843
          ...   
5565    0.000000
5566    0.500000
5567    0.000000
5568    2.333333
5569    0.000000
Name: squalidade, Length: 5570, dtype: float64

In [10]:
df_ordenado = df_sif.sort_values(by=["squalidade"], axis=0, ascending=False)
df_ordenado[["squalidade", "sadquirida", "sgestante"]]

squalidade  sadquirida  sgestante
4798         inf           2          0
732          inf          12          0
3722         inf           4          0
718          inf           1          0
3721         inf           2          0
...          ...         ...        ...
5539         NaN           0          0
5542         NaN           0          0
5544         NaN           0          0
5552         NaN           0          0
5563         NaN           0          0

[5570 rows x 3 columns]

**Explorei a tabela deste indicador de qualidade em ordem descrescente e percebi que dos 5570 municípios, já no 864º município, o valor da razão seria de 5 e no 1939º esse valor seria de 2.5, valores estes que são absolutamente incompatíveis com qualquer sistema de notificação que esteja funcionando. Não faz sentido acreditar que em 84% dos municípios brasileiros, para cada 3 casos de sífilis em mulheres teremos 1 em grávida e 2 em não grávidas. Ainda pior, em 65% dos municípios brasileiros, teríamos mais casos de sífilis em mulheres gestantes do que em mulheres não gestantes. Considero aqui que a razão entre os casos notificados de sífilis em homens e mulheres não grávidas é 3:2 (pesquisei no DATASUS, valor aproximado). Desconsidero nesta análise que o número de casos de sífilis em grávidas inclui somente casos confirmados, o que só contribui para aumentar a certeza da baixa qualidade dos dados. Este é o momento mais crítico do trabalho porque comprova a principal hipótese do mesmo, a de que temos problemas muito graves em notificações de doenças e agravos. Continuo a exploração dos dados de sífilis mas considero que não é necessário continuar explorando mais doenças ou agravos pois nenhuma delas teria uma forma de validar interna tão forte quanto a desta doença que separou grávidas de não grávidas. Ainda, os modelos de machine learning serão realizados, mas fica claro que devem ser interpretados com extrema cautela ou nem interpretados.**

Quantificados os valores infinitos e nulos.

In [11]:
df_sif[df_sif["squalidade"] == float("inf")]["squalidade"].count()

382

In [12]:
df_sif["squalidade"].isna().sum()

187

Retirados valores infinitos e nulos

In [13]:
df_sif.replace([np.inf, -np.inf], np.nan, inplace=True)

In [14]:
df_sif = df_sif.dropna()

In [15]:
df_sif

codigo  aids  animais  dengue  hanseniase  hepatites  sadquirida  \
0     110001     1      192     833          51         41          28   
1     110037     1      199      90          86         12          13   
2     110040     5       62      65          43         66          18   
3     110034     5       78     126          24          8          16   
4     110002   121      252     198         224        240         296   
...      ...   ...      ...     ...         ...        ...         ...   
5565  522145     0       24     115          12          0           0   
5566  522157     0       17     137          11          1           1   
5567  220460     0        0      14           6          2           0   
5568  220590     0        0      21           5          0           7   
5569  250523     0        0      33           4          1           0   

      scongenita  sgestante  toxico  ...        pib  per_capita  populacao  \
0              2         12     186  ...   495629.0    21600.73      22945   
1              1          2      31  ...   288681.0    21802.07      13241   
2              2         11      18  ...   358464.0    16728.75      21428   
3              1          7      35  ...   260348.0    18065.91      14411   
4              8         89     321  ...  2578830.0    23908.38     107863   
...          ...        ...     ...  ...        ...         ...        ...   
5565           0          1       3  ...    51875.0    14821.32       3500   
5566           1          2       0  ...    47960.0    16804.54       2854   
5567           1          2       1  ...    30136.0     7773.10       3877   
5568           3          3       3  ...    51080.0     9549.44       5349   
5569           0          1       0  ...    60564.0     9533.11       6353   

      latitude  longitude  regiao nascimentos mortalidade_geral  \
0    -11.92870   -61.9958       N        1828               659   
1    -12.14050   -61.8379       N         803               252   
2     -9.71470   -63.3193       N        1154               369   
3    -11.34670   -62.2852       N         911               414   
4     -9.90612   -63.0330       N        8580              2557   
...        ...        ...     ...         ...               ...   
5565 -13.50830   -48.7421      CO         155               113   
5566 -14.28400   -49.9205      CO         120                88   
5567  -5.98900   -42.5601      NE         230               115   
5568  -8.00434   -43.8723      NE         344               148   
5569  -6.91334   -35.2505      NE         477               239   

     mortalidade_infantil squalidade  
0                      23   2.333333  
1                       4   6.500000  
2                      24   1.636364  
3                       8   2.285714  
4                     111   3.325843  
...                   ...        ...  
5565                    2   0.000000  
5566                    1   0.500000  
5567                    1   0.000000  
5568                    5   2.333333  
5569                    5   0.000000  

[5001 rows x 35 columns]

Definido que somente cidades com mais de 5 registros de caso em gestantes seriam analisadas.

In [16]:
df_sif = df_sif[df_sif["sgestante"]>=5]

In [17]:
df_sif

codigo  aids  animais  dengue  hanseniase  hepatites  sadquirida  \
0     110001     1      192     833          51         41          28   
2     110040     5       62      65          43         66          18   
3     110034     5       78     126          24          8          16   
4     110002   121      252     198         224        240         296   
5     110045     9      156     721          81         98         134   
...      ...   ...      ...     ...         ...        ...         ...   
5525  510735     0       30     266          21          4          17   
5530  520085     0       11     310          14          1           9   
5531  520145     0       12      75           4          0           6   
5535  520400     0      386     153           3          2          15   
5543  520915     0      222     167          29          2          12   

      scongenita  sgestante  toxico  ...        pib  per_capita  populacao  \
0              2         12     186  ...   495629.0    21600.73      22945   
2              2         11      18  ...   358464.0    16728.75      21428   
3              1          7      35  ...   260348.0    18065.91      14411   
4              8         89     321  ...  2578830.0    23908.38     107863   
5              6         36     105  ...   726031.0    18309.15      39654   
...          ...        ...     ...  ...        ...         ...        ...   
5525           4         12       6  ...   315191.0    56334.34       5595   
5530           2         10       4  ...    75956.0    12429.35       6111   
5531           2          5       4  ...    84493.0    34152.43       2474   
5535           1          7      35  ...   257500.0    32215.68       7993   
5543           0          8      11  ...   189771.0    32175.44       5898   

      latitude  longitude  regiao nascimentos mortalidade_geral  \
0    -11.92870   -61.9958       N        1828               659   
2     -9.71470   -63.3193       N        1154               369   
3    -11.34670   -62.2852       N         911               414   
4     -9.90612   -63.0330       N        8580              2557   
5    -10.20600   -63.8312       N        3064               747   
...        ...        ...     ...         ...               ...   
5525 -10.80190   -52.7351      CO         487               111   
5530 -16.25300   -49.9799      CO         359               207   
5531 -18.29450   -51.1520      CO         155                78   
5535 -15.80620   -46.9237      CO         529               199   
5543 -18.44100   -50.1360      CO         243               156   

     mortalidade_infantil squalidade  
0                      23   2.333333  
2                      24   1.636364  
3                       8   2.285714  
4                     111   3.325843  
5                      39   3.722222  
...                   ...        ...  
5525                    6   1.416667  
5530                    7   0.900000  
5531                    1   1.200000  
5535                    8   2.142857  
5543                    7   1.500000  

[3214 rows x 35 columns]

Definido que somente cidades com mais casos de sífilis em não gestantes do que em mulheres gestantes seriam analisados.

In [18]:
df_sif = df_sif[df_sif["sadquirida"]>df_sif["sgestante"]]

In [19]:
df_sif

codigo  aids  animais  dengue  hanseniase  hepatites  sadquirida  \
0     110001     1      192     833          51         41          28   
2     110040     5       62      65          43         66          18   
3     110034     5       78     126          24          8          16   
4     110002   121      252     198         224        240         296   
5     110045     9      156     721          81         98         134   
...      ...   ...      ...     ...         ...        ...         ...   
5509  500625     0       10     232           7          6          19   
5525  510735     0       30     266          21          4          17   
5531  520145     0       12      75           4          0           6   
5535  520400     0      386     153           3          2          15   
5543  520915     0      222     167          29          2          12   

      scongenita  sgestante  toxico  ...        pib  per_capita  populacao  \
0              2         12     186  ...   495629.0    21600.73      22945   
2              2         11      18  ...   358464.0    16728.75      21428   
3              1          7      35  ...   260348.0    18065.91      14411   
4              8         89     321  ...  2578830.0    23908.38     107863   
5              6         36     105  ...   726031.0    18309.15      39654   
...          ...        ...     ...  ...        ...         ...        ...   
5509           2         10       7  ...   178414.0    46778.58       3814   
5525           4         12       6  ...   315191.0    56334.34       5595   
5531           2          5       4  ...    84493.0    34152.43       2474   
5535           1          7      35  ...   257500.0    32215.68       7993   
5543           0          8      11  ...   189771.0    32175.44       5898   

      latitude  longitude  regiao nascimentos mortalidade_geral  \
0    -11.92870   -61.9958       N        1828               659   
2     -9.71470   -63.3193       N        1154               369   
3    -11.34670   -62.2852       N         911               414   
4     -9.90612   -63.0330       N        8580              2557   
5    -10.20600   -63.8312       N        3064               747   
...        ...        ...     ...         ...               ...   
5509 -22.65570   -53.8577      CO         391               127   
5525 -10.80190   -52.7351      CO         487               111   
5531 -18.29450   -51.1520      CO         155                78   
5535 -15.80620   -46.9237      CO         529               199   
5543 -18.44100   -50.1360      CO         243               156   

     mortalidade_infantil squalidade  
0                      23   2.333333  
2                      24   1.636364  
3                       8   2.285714  
4                     111   3.325843  
5                      39   3.722222  
...                   ...        ...  
5509                    3   1.900000  
5525                    6   1.416667  
5531                    1   1.200000  
5535                    8   2.142857  
5543                    7   1.500000  

[2054 rows x 35 columns]

In [20]:
df_ordenado = df_sif.sort_values(by=["squalidade"], axis=0, ascending=False)
correlacao = df_ordenado[["squalidade", "sadquirida", "sgestante", "populacao", "estado",
                          "per_capita", "estabelecimentos", "profissionais"]]
correlacao.corr().style.background_gradient(axis=None)

Identificada a necessidade de calcular notificações e número de profissionais e estabelecimentos pela população.
Calculando número de pessoas notificadas com a doença por 100 mil habitantes nos 5 anos estudados

In [21]:
df_sif["radquirida"] = 100000*df_sif["sadquirida"]/df_sif["populacao"]
df_sif["rgestante"] = 100000*df_sif["sgestante"]/df_sif["populacao"]
df_sif["rprofissionais"] = 100000*df_sif["profissionais"]/df_sif["populacao"]
df_sif["restabelecimentos"] = 100000*df_sif["estabelecimentos"]/df_sif["populacao"]
df_sif

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documen

codigo  aids  animais  dengue  hanseniase  hepatites  sadquirida  \
0     110001     1      192     833          51         41          28   
2     110040     5       62      65          43         66          18   
3     110034     5       78     126          24          8          16   
4     110002   121      252     198         224        240         296   
5     110045     9      156     721          81         98         134   
...      ...   ...      ...     ...         ...        ...         ...   
5509  500625     0       10     232           7          6          19   
5525  510735     0       30     266          21          4          17   
5531  520145     0       12      75           4          0           6   
5535  520400     0      386     153           3          2          15   
5543  520915     0      222     167          29          2          12   

      scongenita  sgestante  toxico  ...  longitude  regiao  nascimentos  \
0              2         12     186  ...   -61.9958       N         1828   
2              2         11      18  ...   -63.3193       N         1154   
3              1          7      35  ...   -62.2852       N          911   
4              8         89     321  ...   -63.0330       N         8580   
5              6         36     105  ...   -63.8312       N         3064   
...          ...        ...     ...  ...        ...     ...          ...   
5509           2         10       7  ...   -53.8577      CO          391   
5525           4         12       6  ...   -52.7351      CO          487   
5531           2          5       4  ...   -51.1520      CO          155   
5535           1          7      35  ...   -46.9237      CO          529   
5543           0          8      11  ...   -50.1360      CO          243   

      mortalidade_geral  mortalidade_infantil  squalidade  radquirida  \
0                   659                    23    2.333333  122.030944   
2                   369                    24    1.636364   84.002240   
3                   414                     8    2.285714  111.026299   
4                  2557                   111    3.325843  274.422184   
5                   747                    39    3.722222  337.923034   
...                 ...                   ...         ...         ...   
5509                127                     3    1.900000  498.164657   
5525                111                     6    1.416667  303.842717   
5531                 78                     1    1.200000  242.522231   
5535                199                     8    2.142857  187.664206   
5543                156                     7    1.500000  203.458800   

       rgestante rprofissionais restabelecimentos  
0      52.298976     296.360863        152.538679  
2      51.334702     121.336569         88.669031  
3      48.574006     208.174311        152.661162  
4      82.512076     351.371647        114.960645  
5      90.785293     196.701468         83.219852  
...          ...            ...               ...  
5509  262.191924     498.164657        131.095962  
5525  214.477212     250.223414        178.731010  
5531  202.101859     606.305578        121.261116  
5535   87.576630     337.795571        100.087577  
5543  135.639200     220.413700         84.774500  

[2054 rows x 39 columns]

In [22]:
df_ordenado = df_sif.sort_values(by=["squalidade"], axis=0, ascending=False)
correlacao = df_ordenado[["squalidade", "radquirida", "rgestante", "rprofissionais", 
                          "restabelecimentos", "populacao", "per_capita",]]
correlacao.corr().style.background_gradient(axis=None)

Não existe correlação entre o número de casos confirmados de sífilis em gestantes e o número de profissionais, estabelecimentos e renda per capita de um município (<10%). Existe uma baixa correlação entre o número de casos notificados de sífilis em adultos não gestantes e o número de profissionais, estabelecimentos e renda per capita de um município (17.5 a 25%). Essa diferença não muito significativa de valores é matematicamente pequena. Porém expressa com clareza a ideia de que o sistema de notificação de sífilis em gestantes funciona relativamente bem independente do município brasileiro, enquanto a notificação de sífilis em adultos não gestantes é muito ruim. Não faz sentido acreditar que onde existem mais profissionais de saúde, mais estabelecimentos de saúde e mais riqueza, é justamente onde temos mais doenças. Acredito que esse resultado pode ser extrapolado para outros agravos e doenças.